src/hipscat/io/write_metadata.py

```diff
@@ -57,6 +59,32 @@ def write_catalog_info(args, histogram):
     write_json_file(metadata, metadata_filename)
 
 
+def write_provenance_info(args, tool_args):
+    """Write a provenance_info.json file with all assorted catalog creation metadata
+
+    Args:
+        args (:obj:`PartitionArguments`): collection of runtime arguments for the partitioning job
+        tool_args (:obj:`dict`): dictionary of additional arguments provided by the tool creating
+            this catalog.
+    """
+    metadata = {}
+    metadata["catalog_name"] = args.catalog_name
+    metadata["version"] = version("hipscat")
+    now = datetime.now()
+    metadata["generation_date"] = now.strftime("%Y.%m.%d")
+    metadata["epoch"] = args.epoch
+    metadata["ra_kw"] = args.ra_column
+    metadata["dec_kw"] = args.dec_column
+    metadata["id_kw"] = args.id_column
+
+    metadata["origin_healpix_order"] = args.highest_healpix_order
+    metadata["pixel_threshold"] = args.pixel_threshold
+
+    metadata_filename = os.path.join(args.catalog_path, "provenance_info.json")
```

Comment:
Could this file name be a constant?

Reply:
I might do a follow-up to clean up constants for other files.